### PR TITLE
Update manual testing of DG

### PR DESCRIPTION
**Pull request by xyT-T**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1228,27 +1200,15 @@ testers are expected to do more *exploratory* testing.
...
-   1. Test case: `delete_contact 1`<br>
-      Expected: First contact in the event is deleted from the entire list. Returns back to home page, with the full list showing that the contact has been deleted. Details of the deleted contact shown in the status message.
-
-   1. Test case: `delete_contact 0`<br>
-      Expected: No contact is deleted. Error details shown in the status message.
+   1. Test cases and expected outcomes are similar to the previous case except for that the index is based on the currently displayed contact list associated to the selected event.
 
-   1. Other incorrect delete commands to try: `delete_contact`, `delete_contact x`, `...` (where x is larger than the list size)<br>
-      Expected: Similar to previous.
+4. Deleting a contact on other pages causes an error since the contact list is not being displayed.
```
Comment (vijay-shankaranand) [inline docs/DeveloperGuide.md:1211]: Perhaps change this to a note box instead?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1269,74 +1229,46 @@ testers are expected to do more *exploratory* testing.
...
-   1. Test case: `find_contact alex`<br>
-      Expected: All contacts in the currently filtered list with the name containing `alex` is filtered and shown. Command success status message shown with the number of contacts listed (can be 0 if no valid contacts found).
+   1. Test cases and expected outcomes are similar to the previous case except for that only contacts in the currently displayed contact list linked to the selected event will be filtered.
 
-   1. Test case: `find_contact`<br>
-      Expected: Error details shown in the status message.
-
-   1. Other incorrect delete commands to try: `find_contact @`, `...`<br>
-      Expected: Similar to previous.
+4. Finding contacts on other pages causes an error since the contact list is not being displayed.
```
Comment (vijay-shankaranand) [inline docs/DeveloperGuide.md:1267]: Same as previous comment

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1400,31 +1304,36 @@ testers are expected to do more *exploratory* testing.
...
-      Expected: Deletes the `clients` tag. Returns back to home page, and reflects the changes in the entire contact list. Details of the deleted tag shown in the status message.
+   1. Test cases and expected outcomes are similar to the previous case.
 
-   1. Test case: `delete_tag t/`<br>
-      Expected: No tag is deleted. Error details shown in the status message.
+3. Filtering contacts after selecting an event
 
-   1. Other incorrect delete tag commands to try: `delete_tag`, `delete_tag x`, `...`<br>
-      Expected: Similar to previous.
+   1. Prerequisites: Be on the selected event page using the `select_event x` command in which x represents a positive integer that is smaller than or equal to the event list size. Multiple contacts in the list of contacts linked to the event.
```
Comment (vijay-shankaranand) [inline docs/DeveloperGuide.md:1332]: perhaps could state x in `select_event x` refers to index of the list displayed instead?

Comment (vijay-shankaranand) [reply docs/DeveloperGuide.md:1332]:
```suggestion
   1. Prerequisites: Be on the selected event page using the `select_event x` command in which x represents a positive integer that is smaller than or equal to the event list size. There must be contacts linked to the event.
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1400,31 +1304,36 @@ testers are expected to do more *exploratory* testing.
...
+
+   1. Test cases and expected outcomes are similar to the previous case except for that only contacts in the currently displayed contact list linked to the selected event will be filtered.
+
+4. Filtering contacts on other pages causes an error since the contact list is not being displayed.
 
 ### Adding an event
```
Comment (vijay-shankaranand) [inline docs/DeveloperGuide.md:1339]: perhaps could write here that adding of event can be done on any page to sync with other features

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1458,27 +1367,34 @@ testers are expected to do more *exploratory* testing.
...
+
+   3. Test case: `link ev/NUS Career Fest c/Alice Black c/David Li`<br>
+      Expected: Contact `Alice Black` and contact `David Li` are linked to the event `NUS Career Fest`. Command success status message shown.
+
+   4. Test case: `link`<br>
+     Expected: Error details shown in the status message with no operation performed.
+
+### Unlinking contacts from an event
+
+Similar to `link` command except for that the command word is `unlink` and the expected result is opposite.
```
Comment (vijay-shankaranand) [inline docs/DeveloperGuide.md:1397]:
```suggestion
Similar to `link` command except that the command word is `unlink` and the expected result is opposite.
```

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -1511,63 +1427,74 @@ testers are expected to do more *exploratory* testing.
...
-   1. Test case: `select_event 1`<br>
-      Expected: Selects the first event in the list. Displays the event page, displaying the contacts linked to the event and the tasks of the event. Details of the selected event shown in the status message.
+   1. Test cases and expected outcomes are similar to the previous case.
 
-   1. Test case: `select_event 0`<br>
-      Expected: No event is selected. Error details shown in the status message.
-
-   1. Other incorrect select event commands to try: `select_event`, `select_event x`, `...` (where x is an integer larger than the list size)<br>
-      Expected: Similar to previous.
+4. Selecting~~~~ an event on other pages causes an error since the event list is not being displayed.
```
Comment (vijay-shankaranand) [inline docs/DeveloperGuide.md:1438]:
```suggestion
4. Selecting an event on other pages causes an error since the event list is not being displayed.
```

Comment (vijay-shankaranand) [reply docs/DeveloperGuide.md:1438]: perhaps include for selecting event when page is on view_events?

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -1511,63 +1427,74 @@ testers are expected to do more *exploratory* testing.
...
+
+   3. Other incorrect add task commands to try: `mark_task`, `mark_task ev/NUS Career Fair 2024`<br>
+      Expected: Similar to previous.
+
+### Unmarking a task
+
+1. A task can be unmarked from any page.
+
+   1. Test case: `unmark_task td/Book Venue ev/NUS Career Fair 2024`<br>
+      Expected: Book Venue task of NUS Career Fair 2024 event is marked as not completed. Command success status message shown. Goes the respective event page and displays the tasks of the event on the right.
```
Comment (vijay-shankaranand) [inline docs/DeveloperGuide.md:1490]:
```suggestion
      Expected: Book Venue task of NUS Career Fair 2024 event is marked as not completed. Command success status message shown. Goes to the respective event page and displays the tasks of the event on the right.
```

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -1511,63 +1427,74 @@ testers are expected to do more *exploratory* testing.
...
 **Note:** If the task has a due date in the next 3 days or is overdue, then the task will appear under the 'Task Due' section of the home page.
 
 </box>
 
+### Marking a task
+
+1. A task can be marked from any page.
+
+   1. Test case: `mark_task td/Book Venue ev/NUS Career Fair 2024`<br>
+      Expected: Book Venue task of NUS Career Fair 2024 event is marked as completed. Command success status message shown. Goes the respective event page and displays the tasks of the event on the right.
```
Comment (vijay-shankaranand) [inline docs/DeveloperGuide.md:1477]:
```suggestion
      Expected: Book Venue task of NUS Career Fair 2024 event is marked as completed. Command success status message shown. Goes to the respective event page and displays the tasks of the event on the right.
```

**Review by vijay-shankaranand — CHANGES_REQUESTED** [review]
Some considerations to ponder over. Great effort minimizing duplicate sections under manual testing !

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (xyT-T) [reply docs/DeveloperGuide.md:1211]: good suggestion!

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -1158,45 +1070,17 @@ testers are expected to do more *exploratory* testing.
...
-1. Adding a contact while all contacts are being shown
-
-   1. Prerequisites: List all contacts using the `view_contacts` command. Multiple contacts in the list.
+1. A contact can be added from any page.
 
    1. Test case: `add_contact n/John Doe p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 t/vendors t/clients`<br>
       Expected: Contact named John Doe with the respective details created, provided that there is no contact named John Doe or has the phone number 98765432. Returns back to home page.
 
-   1. Test case: `add_contact n/John  Doe p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 t/vendors t/clients`<br>
+   1. Test case: `add_contact n/John &nbsp;&nbsp; Doe p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 t/vendors t/clients`<br>
```
Comment (rionshocker) [inline docs/DeveloperGuide.md]: It is printed as `&nbsp;`.  Maybe we can find a different way to show the double whitespace.

**Review by rionshocker — APPROVED** [review]
LGTM!

**Pull request merged**